pySDC/projects/parallelSDC_reloaded/nilpotency.py

```diff
@@ -36,7 +36,7 @@ def nilpotencyNS(d, Q):
 nil_MIN_SR_NS = []
 nNodes = range(2, 20)
 for m in nNodes:
-    s = Sweeper({"num_nodes": m, "quad_type": quadType, "node_type": nodeType})
+    s = Sweeper({"num_nodes": m, "quad_type": quadType, "node_type": nodeType}, None)
```

Comment:
Wait ... why can't the base `Sweeper` class have `None` as default value for the `level` argument ?

Reply:
Well, the point of this PR is precisely to force passing of a level during instantiation of the sweeper.

Currently, sweepers are only instantiated without a level in a couple of places to setup the corresponding collocation problem.
Outside of testing a sweeper, you either want to have a level with your sweeper to do timestepping or you want to use qmat to generate the collocation problem. I don't see a situation where instantiating a sweeper without a level is useful outside of testing that the sweeper sets up the collocation problem correctly.

Reply:
Why couldn't we use the Sweeper without any level to simply do some SDC sweeps ? If the level is just the object storing the all solution values at each nodes, then it could be generated by default if the user don't specify it.

To fully speak my mind, I don't like adding positional constructor parameters in general (without default value) when they aren't necessary to define the behavior of the object. It's not very user friendly, and in this case this level parameter seems more related to ML stuff than the sweeper.

Reply:
Ok thanks ! It's clearer now (should have read daniel's issue first, I guess ... sorry :sweat_smile:).

But I'm still puzzled on this : 
- `Level` needs a `Problem` right, but need also to know the number of nodes from the sweeper (e.g for the RK sweeper)
- `Sweeper` will be instantiated with the number of nodes, but also need `Level` object

Feels like a circular dependency, right ? Doesn't help the user side ...

Reply:
Yes, I agree that this is a bit confusing. However, the user does not typically interact with the level. I have never seen any but the core level class being used. So the idea is that users now know more explicitly that the sweeper relies on the level but need not concern themselves with how the level class is implemented.

Reply:
Ok, then let's keep it that way for the PR, it's fine enough. But we'll have to find a better solution, eventually ...